code/modules/reagents/chemistry/holder.dm

```diff
@@ -52,7 +92,20 @@
 	var/last_tick = 1
 	var/addiction_tick = 1
 	var/list/datum/reagent/addiction_list = new/list()
+	///cached list of reagents typepaths (not object references), this is a lazylist for optimisation
+	var/list/datum/reagent/previous_reagent_list
 	var/flags
+	///UI lookup stuff
+	///Keeps the id of the reaction displayed in the ui
+	var/ui_reaction_id = null
+	///Keeps the id of the reagent displayed in the ui
+	var/ui_reagent_id = null
+	///The bitflag of the currently selected tags in the ui
+	var/ui_tags_selected = NONE
+	///What index we're at if we have multiple reactions for a reagent product
+	var/ui_reaction_index = 1
+	///If we're syncing with the beaker - so return reactions that are actively happening
+	var/ui_beaker_sync = FALSE
```

Comment:
The UI for searching reagents should 100% be its own datum rather than stored on all reagents. It could probably be a singleton UI too.